core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1555,22 +1555,41 @@ class WorkspaceService(protected val userInfo: UserInfo,
 
   def createSubmission(workspaceName: WorkspaceName, submissionRequest: SubmissionRequest): Future[SubmissionReport] = {
     for {
-      (workspaceContext, submissionParameters, workflowFailureMode, header) <- prepareSubmission(workspaceName, submissionRequest)
-      submission <- saveSubmission(workspaceContext, submissionRequest, submissionParameters, workflowFailureMode, header)
+      (workspaceContext, submissionId, submissionParameters, workflowFailureMode, header, outputPath) <- prepareSubmission(workspaceName, submissionRequest)
+      submission <- saveSubmission(workspaceContext, submissionId, submissionRequest, outputPath, submissionParameters, workflowFailureMode, header)
     } yield {
       SubmissionReport(submissionRequest, submission.submissionId, submission.submissionDate, userInfo.userEmail.value, submission.status, header, submissionParameters.filter(_.inputResolutions.forall(_.error.isEmpty)))
     }
   }
 
   private def prepareSubmission(workspaceName: WorkspaceName, submissionRequest: SubmissionRequest):
-  Future[(Workspace, Stream[SubmissionValidationEntityInputs], Option[WorkflowFailureMode], SubmissionValidationHeader)] = {
+  Future[(Workspace, UUID, Stream[SubmissionValidationEntityInputs], Option[WorkflowFailureMode], SubmissionValidationHeader, String)] = {
+
+    val submissionId: UUID = UUID.randomUUID()
+
     for {
+      workspaceContext <- getWorkspaceContextAndPermissions(workspaceName, SamWorkspaceActions.write)
+
       _ <- requireComputePermission(workspaceName)
 
+      outputPath = submissionRequest.outputPath match {
+        case None => s"gs://${workspaceContext.bucketName}/${submissionId.toString}"
+        case Some(path) => {
+          parseGcsPath(path) match {
+            case Left(error) => throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, s"The specified outputPath was invalid. ${error.value}"))
+            case Right(GcsPath(GcsBucketName(bucketName), GcsObjectName(objectName, _))) =>
+              if(!bucketName.equalsIgnoreCase(s"${workspaceContext.bucketName}")) throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, s"The specified outputPath must be within the workspace bucket gs://${workspaceContext.bucketName}"))
+              if(!objectName.matches(gcsObjectNameRegex.regex)) throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, s"The specified outputPath was invalid. See https://cloud.google.com/storage/docs/naming-objects for naming guidelines."))
+
+              val strippedPath = path.stripSuffix("/")
+              s"${strippedPath}/${submissionId.toString}"
+          }
+        }
```

Comment:
@nawatts I've added some validation here. It's not 100% rock-solid, (i.e. it doesn't protect against extremely niche things like the acme-challenge case and may be slightly too restrictive with it's character set), but it should catch all of the discouraged/banned characters that would result in actual errors on Google's side

This block also takes into account your feedback about ensuring that the bucket name specified doesn't just begin with the bucket name, and also it strips off trailing slashes to avoid confusion.